esp-hal/src/clock/mod.rs

```diff
@@ -386,10 +381,12 @@ impl<'d> ClockControl<'d> {
     ) -> ClockControl<'d> {
         // like NuttX use 40M hardcoded - if it turns out to be a problem
         // we will take care then
-        #[cfg(feature = "xtal-40mhz")]
-        let xtal_freq = XtalClock::RtcXtalFreq40M;
-        #[cfg(feature = "xtal-26mhz")]
-        let xtal_freq = XtalClock::RtcXtalFreq26M;
+        let xtal_freq = if RtcClock::estimate_xtal_frequency() > 33 {
```

Comment:
For esp32, we also have `XtalClock::RtcXtalFreq24M` - do we need to handle this too?

Reply:
I've never seen one in the wild, so I'd rather just pretend they don't exist to keep things simple for now.

Reply:
Should we just remove that enum variant then? I've never seen one either, and I'd be happy to never see one too :D

Reply:
Probably a good idea, yeah